pkg/pull/archive.go

```diff
@@ -0,0 +1,97 @@
+package pull
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/mholt/archiver"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/base"
+	"github.com/replicatedhq/kots/pkg/upstream"
+	"github.com/replicatedhq/kots/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func writeArchiveAsConfigMap(pullOptions PullOptions, u *upstream.Upstream, baseDir string) error {
+	// Package this app into a bundle so that the Admin Console can write it as the first version...
+	tarGz := archiver.TarGz{
+		Tar: &archiver.Tar{
+			ImplicitTopLevelFolder: true,
+		},
+	}
+
+	paths := []string{
+		path.Join(pullOptions.RootDir, u.Name, "upstream"),
+		path.Join(pullOptions.RootDir, u.Name, "base"),
+		path.Join(pullOptions.RootDir, u.Name, "overlays"),
+	}
+
+	tempDir, err := ioutil.TempDir("", "kots")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp dir")
+	}
+	defer os.RemoveAll(tempDir)
+
+	if err := tarGz.Archive(paths, path.Join(tempDir, "kots-uploadable-archive.tar.gz")); err != nil {
+		return errors.Wrap(err, "failed to create tar gz")
+	}
+
+	archive, err := ioutil.ReadFile(path.Join(tempDir, "kots-uploadable-archive.tar.gz"))
+	if err != nil {
+		return errors.Wrap(err, "failed to read temp file")
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(archive)
+
+	// well.
+	//
+	// let's write this encoded value to a config map with a known label
+	// so that kotsadm will upload it and ingest it as an app
+	// it's really the only way we can get the archive
+	// but etcd and config maps are limited to 1 mb
+	// so let's split it across multiple, if it's larger than 1 mb
+	encodedParts, err := util.SplitStringOnLen(encoded, 756*1000)
```

Comment:
```suggestion
	// 768 * 1000 is sufficiently less than a megabyte to ensure we
	// won't have problems
	encodedParts, err := util.SplitStringOnLen(encoded, 768*1000)
```
moar comment!
Also using a round number (`768`) instead of some random number (`756`)
`(log(768)-log(3))/log(2)=8`

Reply:
ha!